cmd/tetra/tracingpolicy/tracingpolicy.go

```diff
@@ -97,6 +108,21 @@ func addTracingPolicy(ctx context.Context, client tetragon.FineGuidanceSensorsCl
 	}
 }
 
+func deleteTracingPolicy(ctx context.Context, client tetragon.FineGuidanceSensorsClient, yamlFname string) {
+	yamlb, err := os.ReadFile(yamlFname)
+	if err != nil {
+		fmt.Printf("failed to read yaml file %s: %s\n", yamlFname, err)
+		return
+	}
+
+	_, err = client.DeleteTracingPolicy(ctx, &tetragon.DeleteTracingPolicyRequest{
+		Yaml: string(yamlb),
```

Comment:
It seems that the `DeleteTracingPolicyRequest` takes the full yaml as an argument (this is probably my fault). I think it would make more sense to take the name as the argument.

So, basically, redefine `DeleteTracingPolicyRequest` to be:
```
message DeleteTracingPolicyRequest {
	string name = 1;
}
```

What do you think?

Reply:
I think it's very good observation, makes sense. Just one little question is that currently there is some logic for checking whether policy is [namespaced](https://github.com/inliquid/tetragon/blob/pr/inliquid/remove-tracing-policy-method/pkg/server/server.go#L259), and logging namespace. Apart from logging it looks like namespace is not used anywhere. So if we go for deleting tracing policy by name this logging will be removed as well. Is this ok?

Reply:
I think this is perfectly fine, it's still interesting to log the request though, which will only contain the name. It's just a debug trace if we need to see the gRPC handling in action.

Reply:
Keep something like
```go
	logger.GetLogger().WithFields(logrus.Fields{
		"metadata.name":      tp.TpName(),
	}).Debug("Received a DeleteTracingPolicy request")
```

Reply:
Fixed. Please note that "delete <sensor_name>" help string for tracing policy "delete" comment in `tetra` which is in line with sensor's "rm" command which is "rm <sensor_name>" as they effectively use same name on deletion, despite that sensors names are usually something like `gkp-sensor-1`.

Reply:
@mtardy I also squashed commits together as you suggested earlier, not sure if this is necessary.

Reply:
> I think it's very good observation, makes sense. Just one little question is that currently there is some logic for checking whether policy is [namespaced](https://github.com/inliquid/tetragon/blob/pr/inliquid/remove-tracing-policy-method/pkg/server/server.go#L259), and logging namespace. Apart from logging it looks like namespace is not used anywhere. So if we go for deleting tracing policy by name this logging will be removed as well. Is this ok?

Short version: yes that's fine.

Long version: yes, that's fine because currently, we cannot have different policies with the same name, regardless of whether they are namespaced or not.

See https://github.com/cilium/tetragon/blob/f0ae9f0f27f0bfee0164d8996bfb90967fea489b/pkg/sensors/handler.go#L92

This means that the name uniquely identifies the policy (if it exists) so using just the name is fine.

I suspect that, in the future, we might want to change that to allow policies with the same name on different namespaces. When this happens, we can extend the request to include the namespace as well. 

> s in line with sensors "rm" command which is "rm <sensor_name>" as they effectively use same name on deletion

As another note, the sensors are just old interfaces that we should remove. We should only use tracing policies from now on.

Reply:
@kkourt 
> As another note, the sensors are just old interfaces that we should remove. We should only use tracing policies from now on.

We're using sensors to enable/disable policies, which is very useful when you want to just turn on/off some specific channel of information such as monitoring of network activity, as it may generate a lot of load. Would it still be possible to switch tracing policies on/off as easily as with sensors? We would reimplement our "controller" component to consume another gRPC methods if sensors had gone, but can this feature remain please?

Reply:
That's a good point, thanks for mentioning it. We should definitely do that. Created the following issue: https://github.com/cilium/tetragon/issues/1272